src/errors.ts

```diff
@@ -4,15 +4,18 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-
+import { format } from 'node:util';
 /**
  * Any `Error` compatible with the `NamedError` type signature.
  */
-type NamedErrorLike = Error & {
-  readonly name: string;
-  readonly cause?: NamedErrorLike;
-  readonly fullStack?: string;
-};
+type NamedErrorLike =
```

Comment:
widen the type, should be non-breaking.  The "native" Error might not have a name, and now that's allowed as a cause.